recipes/corrade/all/conanfile.py

```diff
@@ -97,54 +101,64 @@ def package(self):
+        self.copy("UseCorrade.cmake", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake"))
+        self.copy("CorradeLibSuffix.cmake", src=share_cmake, dst=os.path.join(self.package_folder, "lib", "cmake"))
+        self.copy(pattern="*.cmake", dst=os.path.join("lib", "cmake"), src=os.path.join(self.source_folder, "cmake"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
-    def _sort_libs(self, correct_order, libs, lib_suffix="", reverse_result=False):
-        # Add suffix for correct string matching
-        correct_order[:] = [s.__add__(lib_suffix) for s in correct_order]
-
-        result = []
-        for expectedLib in correct_order:
-            for lib in libs:
-                if expectedLib == lib:
-                    result.append(lib)
-
-        if reverse_result:
-            # Linking happens in reversed order
-            result.reverse()
-        return result
-
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "Corrade"
         self.cpp_info.names["cmake_find_package_multi"] = "Corrade"
 
-        self.cpp_info.builddirs.append(os.path.join("lib", "cmake"))
-        self.cpp_info.build_modules.append(os.path.join("lib", "cmake", "Corrade", "UseCorrade.cmake"))
-        self.cpp_info.build_modules.append(os.path.join("lib", "cmake", "Corrade", "CorradeLibSuffix.cmake"))
-
-        # See dependency order here: https://doc.magnum.graphics/magnum/custom-buildsystems.html
-        allLibs = [
-            #1
-            "CorradeMain",
-            "CorradeUtility",
-            "CorradeContainers",
-            #2
-            "CorradeInterconnect",
-            "CorradePluginManager",
-            "CorradeTestSuite",
-        ]
-
-        # Sort all built libs according to above, and reverse result for correct link order
         suffix = "-d" if self.settings.build_type == "Debug" else ""
-        builtLibs = tools.collect_libs(self)
-        self.cpp_info.libs = self._sort_libs(allLibs, builtLibs, suffix, True)
-        if self.settings.os == "Linux":
-            self.cpp_info.system_libs = ["m", "dl"]
-
-        self.cpp_info.builddirs = [os.path.join(self.package_folder, "lib", "cmake", "Corrade")]
 
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bindir))
-        self.env_info.PATH.append(bindir)
+        # The FindCorrade.cmake file provided by the library populates some extra stuff
+        self.cpp_info.components["_corrade"].build_modules.append(os.path.join("lib", "cmake", "conan-corrade-vars.cmake"))
+
+        if self.options.with_main:
+            self.cpp_info.components["main"].names["cmake_find_package"] = "Main"
+            self.cpp_info.components["main"].names["cmake_find_package_multi"] = "Main"
+            self.cpp_info.components["main"].names["pkg_config"] = "corrade_main"
```

Comment:
not like this, it suggests an unofficial pkg_config name to others maintenairs

Reply:
but then, on the Conan side, you need to write `corrade::corrade_main` to refer to components from this recipe... I can gather them all at the very end of the recipe and add a comment maybe.

Reply:
Yes, this is why I asked to remove implicit generation of pkg_config files of components on conan side, but it's still there.

Reply:
https://github.com/conan-io/conan/issues/7560

Reply:
wdyt about something like this at the end of the recipe?

```python
        # pkg_config: Add more explicit naming to generated files (avoid filesystem collision).
        for key, cmp in self.cpp_info.components.items():
            self.cpp_info.components[key].names["pkg_config"] = "{}_{}".format(self.name, key)
```

Reply:
It will do for now, I guess.

Reply:
ok